### PR TITLE
Add logging for automatic accession gathering

### LINKED ISSUE #3062: Logging for automatic accession gathering
### Context

We will need some logs around automatic accession gathering and surveying. When any accession gathering script is run, we should know if nothing was returned - it is probably best to just log when it was ran and a count of experiment accessions from each source. 

### Solution or next step

- Create the log stream for accession_gathering
- log when gathering script is ran, include counts from each source

### PR DESCRIPTION
**Pull request by arkid15r**
## Issue Number

Resolves #3062 

## Purpose/Implementation Notes

Add logging for accession agents results.
Remove unused flags.

## Types of changes
- New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: foreman/data_refinery_foreman/gatherer/management/commands/gather_accessions.py [hunk 1]
```diff
@@ -217,28 +201,52 @@ def handle(self, *args, **options):
...
-                int(self.RE_ACCESSION.match(entry.accession_code).group(2)),
+            logger.info(
+                f"Since {options['since']} {agent} has gathered {agent_accessions_count} "
+                f"new accession{pluralize(agent_accessions_count)}."
+            )
+
+        gathered_accessions = sorted(  # Sort the resulting list.
+            (ga for ga in gathered_accessions if self.RE_ACCESSION.match(ga.accession_code)),
+            key=lambda ga: (
+                self.RE_ACCESSION.match(ga.accession_code).group(1),
```
Comment (davidsmejia) [inline foreman/data_refinery_foreman/gatherer/management/commands/gather_accessions.py:219]: Could you add a comment where `RE_ACCESSION` is defined so it is easier to understand.

File: foreman/data_refinery_foreman/gatherer/management/commands/gather_accessions.py [hunk 2]
```diff
@@ -217,28 +201,52 @@ def handle(self, *args, **options):
...
+            if gathered_accessions:
+                output = "\n".join((str(ga) for ga in gathered_accessions))
             else:
-                output = "No accessions found."
+                output = "No new accessions gathered."
             print(output)
         else:
             try:
-                GatheredAccession.objects.bulk_create(entries)
+                GatheredAccession.objects.bulk_create(gathered_accessions)
```
Comment (davidsmejia) [inline foreman/data_refinery_foreman/gatherer/management/commands/gather_accessions.py:245]: I think we should instead do the `bulk_create` in batches and log each successful / unsuccessful batch.

**Review by davidsmejia — CHANGES_REQUESTED** [review]
Some small changes but looks good!

File: foreman/data_refinery_foreman/gatherer/management/commands/gather_accessions.py (hunk 2 above)
Comment (arkid15r) [reply foreman/data_refinery_foreman/gatherer/management/commands/gather_accessions.py]: Done.

**Review by arkid15r — COMMENTED** [review]
PTAL

File: foreman/data_refinery_foreman/gatherer/management/commands/gather_accessions.py [hunk 3]
```diff
@@ -217,28 +206,59 @@ def handle(self, *args, **options):
...
-                self.RE_ACCESSION.match(entry.accession_code).group(1),
-                int(self.RE_ACCESSION.match(entry.accession_code).group(2)),
+            gathered_accessions.update(agent_accessions)
+
+            logger.info(
+                f"Since {options['since']} {agent} has gathered {agent_accessions_count} "
+                f"new accession{pluralize(agent_accessions_count)}."
+            )
+
+        RE_ACCESSION = re.compile(r"(\D+)(\d+)")
```
Comment (davidsmejia) [inline foreman/data_refinery_foreman/gatherer/management/commands/gather_accessions.py:221]:
```suggestion
        # Split into two groups, non-digits and digits ex: (GSE)(1234).
        RE_ACCESSION = re.compile(r"(\D+)(\d+)")
```

Comment (arkid15r) [reply foreman/data_refinery_foreman/gatherer/management/commands/gather_accessions.py:221]: Done!

**Pull request merged**